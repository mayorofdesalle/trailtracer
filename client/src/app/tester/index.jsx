import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
