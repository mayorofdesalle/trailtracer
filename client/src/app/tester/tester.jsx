import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };