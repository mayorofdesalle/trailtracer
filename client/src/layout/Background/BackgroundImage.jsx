import { useCallback } from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';

import Container from '@components/ui/Container';
import checkWebpFeature from '@utils/checkWebpFeature';
import getImageUrl from '@utils/getImageUrl';

const Image = styled(Container)`
    background: url(${({ $image }) => getImageUrl($image)}) repeat;
    opacity: 43%;
`;

/**
 * BackgroundImage
 * @description
 * Default background image.
 * It is used as a fallback for webkit browsers, and as suspense otherwise.
 **/
const BackgroundImage = () => {
    const preference = useSelector(state => state.theme.preference);
    const fileName = preference ? 'backgroundDark' : 'backgroundLight';
    const isWebpSupported = useCallback(() => { checkWebpFeature('lossless', (feature, isSupported) => isSupported); }, []);

    return (
        <Image $image={isWebpSupported ? `${fileName}.webp` : `${fileName}.png`} />
    );
};

export default BackgroundImage;