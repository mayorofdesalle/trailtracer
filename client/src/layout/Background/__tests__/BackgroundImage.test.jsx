import { describe, expect, it, render, screen } from '@app/tester';

import BackgroundImage from '../BackgroundImage';

describe('BackgroundImage', () => {

    it('should render correctly', () => {
        render(<BackgroundImage data-testid='background-image' />);
        expect(screen.getByTestId('background-image')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<BackgroundImage />);
        expect(asFragment()).toMatchSnapshot();
    });

});