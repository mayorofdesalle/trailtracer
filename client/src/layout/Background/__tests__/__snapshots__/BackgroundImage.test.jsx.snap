// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BackgroundImage > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-kFCroH kHwwsn jtmoiL"
  />
</DocumentFragment>
`;
